psi4/src/psi4/libmints/matrix.cc

```diff
@@ -747,6 +747,53 @@ void Matrix::set_column(int h, int m, SharedVector vec)
     }
 }
 
+SharedMatrix Matrix::get_block(Slice rows,Slice cols)
+{
+    const Dimension& rows_start = rows.start();
+    const Dimension& cols_start = cols.start();
+    Dimension block_rows = rows.end() - rows.start();
+    Dimension block_cols = cols.end() - cols.start();
+    SharedMatrix block = std::make_shared<Matrix>("Block",block_rows,block_cols);
+    for (int h = 0; h < nirrep_; h++){
+        int max_p = block_rows[h];
+        int max_q = block_cols[h];
+        for (int p = 0; p < max_p; p++){
+            for (int q = 0; q < max_q; q++){
+                double value = get(h,p + rows_start[h],q + cols_start[h]);
+                block->set(h,p,q,value);
+            }
+        }
+    }
+    return block;
+}
+
+void Matrix::set_block(Slice rows,Slice cols,SharedMatrix block)
+{
+    const Dimension& rows_start = rows.start();
+    const Dimension& cols_start = cols.start();
+    Dimension block_rows = rows.end() - rows.start();
+    Dimension block_cols = cols.end() - cols.start();
+    for (int h = 0; h < nirrep_; h++){
+        int max_p = block_rows[h];
+        int max_q = block_cols[h];
+        for (int p = 0; p < max_p; p++){
+            for (int q = 0; q < max_q; q++){
+                double value = block->get(h,p,q);
+                set(h,p + rows_start[h],q + cols_start[h],value);
+            }
+        }
+    }
+}
+
+/**
+ * Set a matrix block
+ *
+ * @param rows Rows slice
+ * @param cols Columns slice
+ * @return SharedMatrix object
+ */
+void set_block(Slice rows,Slice cols,SharedMatrix block);
+
```

Comment:
For what I can see `View` was not integrated extensively and it is used only in exactly three lines of psi4. View is really a shared pointer plus a set of start/end Dimension objects. It uses pointers to store what are dimension-like objects and should be modernized. Slice here is just a set of start/end Dimension objects. Indeed I think it might be a good idea to write `View` using `Slice` (perhaps my choice of class name is not idea, we could also call it `DimensionRange`, `Range`, etc.). So I think this is not a case of repetition.